File: Boost_Echo_Client/src/AckLogstat.cpp
//
// Created by MAY on 05/01/2022.
//

#include "../include/AckLogstat.h"

AckLogstat::AckLogstat(std::list<AckMessageUserInfo> _userInfos, short _messageOpcode, short _opCode) : AckMessage(_messageOpcode, _opCode) , userInfos(_userInfos){
}

std::string AckLogstat::toString() {
    std::string answer="ACK ";
    answer.append(std::to_string(getMessageOpCode()));
    answer.push_back(' ');
    for(AckMessageUserInfo messageUserInfo : userInfos){
        answer.push_back('\n');
        answer.append(messageUserInfo.toString());
    }

    return answer;
}

File: spl-net/src/main/java/bgu/spl/net/impl/messages/AckLogstat.java
package bgu.spl.net.impl.messages;

import java.util.LinkedList;
import java.util.regex.Pattern;

public class AckLogstat extends AckMessage{
    private LinkedList<AckMessageUserInfo> userInfos;
    public AckLogstat(short messageOpcode, LinkedList<AckMessageUserInfo> userInfos) {
        super(messageOpcode);
        this.userInfos=userInfos;
    }

    public LinkedList<AckMessageUserInfo> getUserInfos(){return userInfos;}
    @Override
    public String toString(){
        short opcode=getOpCode();
        String output=String.valueOf(opcode);
        short messageOpcode=getMessageOpcode();
        output+=messageOpcode;

        LinkedList<AckMessageUserInfo> userinfo=getUserInfos();
        String info="";
        for(int i=0;i<userinfo.size();i++){
            info+=userinfo.get(i).getAge();
            info+=userinfo.get(i).getNumPosts();
            info+=userinfo.get(i).getNumFollowers();
            info+=userinfo.get(i).getNumFollowing();
            info+= "\\|";
        }
        output+=info;

        return output;
    }

}
